app/components/deduped-hearings-list.js

```diff
@@ -0,0 +1,57 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default class DedupedHearingsListComponent extends Component {
+  // the actions "attribute" to display in the hearings list e.g. "ulurpnumber" or 'name'
+  // this is set when the component is rendered
+  attribute = '';
+
+  hearingsSubmitted = false;
+
+  @computed('hearingsSubmitted', 'project')
+  get dedupedHearings() {
+    const dispositions = this.get('project.dispositions');
+    // hearingsSubmitted is set when the component is rendered
+    const { hearingsSubmitted } = this;
+
+    // an array of objects that have been reduced
+    let deduped;
+
+    // setting a new property on each disposition called hearingActions which is an array of objects
+    // this property hearingActions is initally set to an array of the current disposition's action model
+    // during the reduce, if there is a duplicate in the array of dispositions,
+    // the actions model for that duplicate disposition is pushed into this array
+    dispositions.forEach(function(disposition) {
+      disposition.set('hearingActions', [disposition.action]);
+    });
+
+    // hearingsSubmitted is set when the component is rendered
+    if (hearingsSubmitted) {
+      // reduce method to deduplicate dispositions based on location and date of hearing
+      deduped = dispositions.reduce((accumulatedDisps, currentDisp) => {
```

Comment:
nit: this method is doing a lot. I like the idea of returning this reduce by itself from somewhere.

Reply:
@allthesignals "returning this reduce by itself from somewhere" -- what do you mean by that? Separating out these functions?